### PR TITLE
Add Prometheus service monitor for flagger-loadtester

### PR DESCRIPTION
**Pull request by branden**
**What type of PR is this?**

Bug

**What this PR does/ why we need it**:

This adds a service monitor and related service labels so that Prometheus can discover flagger-loadtester's metrics endpoint.

I tested this PR by deploying KBA from this branch, browsing to `/ops/portal/prometheus/targets`, and observing that the flagger-loadtester target is present and available:

Screen Shot 2020-10-30 at 12 10 54 PM

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-72694

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Restores metrics collection for flagger-loadtester.
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

**Review by gracedo — APPROVED** [review]
Would you be able to verify this manually on a cluster that these metrics are getting scraped into prom?

**Comment by branden** [comment]
Looks like https://github.com/mesosphere/charts/pull/860 broke the flagger chart when I changed the CRD resource version without updating the spec to match the new schema. I'll need to fix that before this PR can proceed.

File: addons/flagger/flagger.yaml [hunk 1]
```diff
@@ -8,9 +8,9 @@ metadata:
   annotations:
     sidecar.istio.io/inject: "false"
     appversion.kubeaddons.mesosphere.io/flagger: "0.19.0"
-    catalog.kubeaddons.mesosphere.io/addon-revision: "0.19.0-1"
```
Comment (armandgrillet) [inline addons/flagger/flagger.yaml:11]: If we bump the chartReference.version to `0.20.1`, should we update this and the `appversion.kubeaddons.mesosphere.io/flagger` to `0.20` too?

Comment (branden) [reply addons/flagger/flagger.yaml:11]: I think it was a coincidence that the app and chart version were the same. Chart version 0.20.1 still deploys flagger 0.19.0: https://github.com/mesosphere/charts/blob/659d0ddada7a74e57e6a73f5a0de01fcf633b151/staging/flagger/Chart.yaml#L3-L4

Comment (s12chung) [reply addons/flagger/flagger.yaml:11]: I agree with @branden. `appversion` is the internal app version inside of the chart.

**Review by s12chung — APPROVED** [review]

**Comment by branden** [comment]
The teamcity job is failing because of a bug in the chart that's already present: https://jira.d2iq.com/browse/D2IQ-73316 This bug occurs when flagger is deployed, removed, and deployed again. This PR is triggering it because it's the first change to flagger since we added a set of tests that causes flagger to be deployed a second time to the test cluster.

**Pull request merged**